Updating docs for unimodules

brentvatne opened the pull request:
# Why

Unimodules as a way to use Expo tools makes a lot of the current documentation confusing. This PR includes a bunch of changes to attempt to make the docs work in the world of a managed and a bare workflow.

<img width="1386" alt="screen shot 2019-02-22 at 10 47 11 am" src="https://user-images.githubusercontent.com/90494/53263795-3d689c80-368f-11e9-9b3d-4285201f459a.png">

brentvatne committed:
[docs] Project lifecycle changes and some other cleanup

brentvatne committed:
A bunch of incremental improvements

brentvatne committed:
Make section groups and have them be collapsable

brentvatne committed:
Group API reference too

brentvatne committed:
Shuffle docs a bit for grouping

brentvatne committed:
Tweak styles

brentvatne committed:
More style tweaks

brentvatne committed:
Reorder a few things in fundamentals and fix AppAuth name

brentvatne committed:
Create a bare workflow group

ide commented:
Looks like some links are broken according to CI:
```
Getting links from: http://localhost:8000/versions/latest/sdk/overview/
\├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/latest/sdk/overview.js
├─BROKEN─ http://localhost:8000/versions/latest/sdk/overview/contacts/#getcontactsasync (HTTP_404)
Finished! 32 links found. 30 excluded. 1 broken.

Getting links from: http://localhost:8000/versions/v32.0.0/sdk/overview/
|├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/v32.0.0/sdk/overview.js
├─BROKEN─ http://localhost:8000/versions/v32.0.0/sdk/overview/contacts/#getcontactsasync (HTTP_404)
Finished! 200 links found. 198 excluded. 1 broken.

Getting links from: http://localhost:8000/versions/v32.0.0/distribution/introduction/
├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/v32.0.0/distribution/introduction.js
├─BROKEN─ http://localhost:8000/versions/v32.0.0/distribution/introduction/building-standalone-apps/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/v32.0.0/distribution/distribution/app-stores/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/v32.0.0/distribution/introduction/release-channels/ (HTTP_404)
Finished! 81 links found. 77 excluded. 3 broken.

Getting links from: http://localhost:8000/versions/v32.0.0/expokit/overview/
├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/v32.0.0/expokit/overview.js
├─BROKEN─ http://localhost:8000/versions/v32.0.0/expokit/distribution/building-standalone-apps/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/v32.0.0/expokit/overview/eject/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/v32.0.0/expokit/overview/expokit/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/v32.0.0/expokit/overview/advanced-expokit-topics/ (HTTP_404)
Finished! 82 links found. 77 excluded. 4 broken.

Getting links from: http://localhost:8000/versions/latest/distribution/introduction/
├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/latest/distribution/introduction.js
├─BROKEN─ http://localhost:8000/versions/latest/distribution/introduction/building-standalone-apps/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/latest/distribution/distribution/app-stores/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/latest/distribution/introduction/release-channels/ (HTTP_404)
Finished! 32 links found. 28 excluded. 3 broken.

Getting links from: http://localhost:8000/versions/latest/expokit/overview/
├───OK─── http://localhost:8000/_next/static/hfYsqSTfwIzstJ8Qo8Tg4/pages/versions/latest/expokit/overview.js
├─BROKEN─ http://localhost:8000/versions/latest/expokit/distribution/building-standalone-apps/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/latest/expokit/overview/eject/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/latest/expokit/overview/expokit/ (HTTP_404)
├─BROKEN─ http://localhost:8000/versions/latest/expokit/overview/advanced-expokit-topics/ (HTTP_404)
Finished! 33 links found. 28 excluded. 4 broken.
```

brentvatne committed:
Add consistent sections for installation and api to sdk

brentvatne committed:
Remove expo-core from dependencies

brentvatne committed:
Add expo-module script for generating README

brentvatne committed:
Add some helpful links to footer

brentvatne committed:
Update website in unimodules to point to docs

brentvatne committed:
Update READMEs

brentvatne committed:
Backport docs changes and a few small changes

brentvatne committed:
Backport more docs from unversioned

brentvatne committed:
Fix broken links

brentvatne committed:
Fix some more links

brentvatne commented:
> Looks like some links are broken according to CI:
> (redacted)

thanks, done

brentvatne committed:
Move peerDependencies on unimodule packages to unimodulesPeerDependencies
Also move some direct dependencies on unimodule packages to
unimodulesPeerDependencies. This field is currently unused, but
peerDependencies does not work with transitive dependencies so users
will get a lot of warnings and they won't be particularly helpful. Version
changed to * for now because it's unused and I'm not sure everything is
up to date anyhow.

brentvatne committed:
Add note about scoping in expo-font

brentvatne committed:
Update unimodule packages to 3.0

expo-gl@3.0.0
expo-sms@3.0.0
expo-core@3.0.0
expo-font@3.0.0
expo-print@3.0.0
expo-camera@3.0.0
expo-sensors@3.0.0
expo-location@3.0.0
expo-constants@3.0.0
expo-file-system@3.0.0
expo-permissions@3.0.0
expo-analytics-segment@3.0.0
expo-media-library@3.0.0
expo-font-interface@3.0.0
expo-camera-interface@3.0.0
expo-sensors-interface@3.0.0
expo-constants-interface@3.0.0
expo-react-native-adapter@3.0.0
expo-file-system-interface@3.0.0
expo-permissions-interface@3.0.0
expo-image-loader-interface@3.0.0
expo-face-detector-interface@3.0.0
expo-face-detector@3.0.0
expo-barcode-scanner-interface@3.0.0
expo-barcode-scanner@3.0.0
expo-gl-cpp@3.0.0
expo-asset@3.0.0
expo-module-template@3.0.0

brentvatne committed:
Release 3.0 for a few other packages

brentvatne committed:
Update expo package and publish expo-intent-launcher

brentvatne committed:
Add Hello World doc
